configuration/persistence.md

```diff
@@ -7,12 +7,238 @@ title: Persistence
+In order for items to be persisted across openHAB restarts, you must define a `restoreOnStartup` strategy for all your items.
+Then, whatever state they were in before the restart will be restored automatically.
+The following example persists all items on every change and restores them at startup:
+
+```java
+Strategies {
+  default = everyUpdate
+}
+
+Items {
+  * : strategy = everyChange, restoreOnStartup
+}
+```
+
+## Persistence Extensions in Scripts and rules
+
+To make use of persisted states inside scripts and rules, a few useful extensions have been defined on items.
+In contrast to an action (which is a function that can be called anywhere in a script or rule), an extension is a function that is only available like a method on a certain type.
+This means that the persistence extensions are available like methods on all items.
+
+<!-- TODO: Reword the above section to replace the word "like" with something that is more clear-->
+
+Example:
+
+The statement
+
+`Temperature.historicState(now.minusDays(1))`
+
+will return the state of the item "Temperature" from 24 hours ago.
+You can easily imagine that you can implement very powerful rules using this feature.
+
+Here is the full list of available persistence extensions:
+
+```java
+<item>.persist - Persists the current state
+<item>.lastUpdate - Query for the last update timestamp of a given item.
+<item>.historicState(AbstractInstant) - Retrieves the historic item at a certain point in time
+<item>.changedSince(AbstractInstant) - Checks if the state of the item has (ever) changed since a certain point in time
+<item>.updatedSince(AbstractInstant) - Checks if the state of the item has been updated since a certain point in time
+<item>.maximumSince(AbstractInstant) - Gets the Item with the maximum value (state) since a certain point in time
+<item>.minimumSince(AbstractInstant) - Gets the Item with the minimum value (state) since a certain point in time
+<item>.averageSince(AbstractInstant) - Gets the average value of the state of a given item since a certain point in time.
+<item>.deltaSince(AbstractInstant) - Gets the difference value of the state of a given item since a certain point in time.
+<item>.previousState() - Retrieves the previous item (returns HistoricItem).
+<item>.previousState(true) - Retrieves the previous item, skips items with equal state values and searches the first item with state not equal the current state (returns HistoricItem).
+<item>.sumSince(AbstractInstant) - Retrieves the sum of the previous states since a certain point in time. (OpenHab 1.8)
+```
+
+These extensions use the default persistence service that is configured as the default persistence service.  (Refer to Default Persistence Service above to configure this.)
+
+Note that you can specify that a different persistence service be used with a particular extension.
+Do this by appending a String as an optional additional parameter at the end of the extension (e.g. "rrd4j" or "sense")
```

Comment:
Maybe an example would be good here.

Reply:
I agree, and I was not aware that one could do this.  I have a development system in my office, and if I get time, I can check it out by installing another persistence add-on and then checking that it works as expected.  From what was in the wiki, I believe the syntax would be to add something like ".rrd4j" at the end of the function, but this should be verified before we publish an example.  For now I will add the following:  <!-- TODO:Add an example of this.  I assume it is as simple as adding .rrd4j to the end of one of these, but this should be verified before being published. -->